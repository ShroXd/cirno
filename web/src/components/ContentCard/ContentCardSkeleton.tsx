import { PhotoIcon } from '@heroicons/react/24/outline'
import { CardBody, CardHeader, Typography } from '@material-tailwind/react'

import { ContentCardContainer } from './Container'

interface ContentCardSkeletonProps {
  className?: string
}

export const ContentCardSkeleton = ({
  className,
}: ContentCardSkeletonProps) => (
  <ContentCardContainer className={`animate-pulse ${className}`}>
    <CardHeader
      shadow={false}
      floated={false}
      className='relative m-0 mb-5 grid h-64 place-items-center rounded-b-none bg-gray-300'
    >
      <PhotoIcon className='h-12 w-12 text-gray-500' />
    </CardHeader>
    <CardBody className='pb-4 pt-0'>
      <Typography
        as='div'
        variant='h1'
        className='h-3 rounded-full bg-gray-300'
      >
        &nbsp;
      </Typography>
    </CardBody>
  </ContentCardContainer>
)
