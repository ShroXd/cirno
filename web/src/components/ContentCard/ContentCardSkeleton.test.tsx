import { render } from '@testing-library/react'
import { axe } from 'jest-axe'
import { describe, expect, it } from 'vitest'

import { ContentCardSkeleton } from './ContentCardSkeleton'

describe('ContentCard skeleton', () => {
  it('should render without a11y violations', async () => {
    const { container } = render(<ContentCardSkeleton />)
    expect(await axe(container)).toHaveNoViolations()
  })

  it('should render correctly', () => {
    const { container } = render(<ContentCardSkeleton />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render correctly with custom className', () => {
    const { container } = render(<ContentCardSkeleton className='mt-4' />)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
