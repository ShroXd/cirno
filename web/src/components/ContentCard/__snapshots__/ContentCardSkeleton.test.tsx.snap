// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentCard skeleton > should render correctly 1`] = `
<div
  class="relative bg-clip-border text-gray-700 flex select-none flex-col overflow-hidden rounded-xl bg-white shadow-lg animate-pulse undefined"
>
  <div
    class="bg-clip-border rounded-xl overflow-hidden text-gray-700 relative m-0 mb-5 grid h-64 place-items-center rounded-b-none bg-gray-300"
  >
    <svg
      aria-hidden="true"
      class="h-12 w-12 text-gray-500"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m2.25 15.75 5.159-5.159a2.25 2.25 0 0 1 3.182 0l5.159 5.159m-1.5-1.5 1.409-1.409a2.25 2.25 0 0 1 3.182 0l2.909 2.909m-18 3.75h16.5a1.5 1.5 0 0 0 1.5-1.5V6a1.5 1.5 0 0 0-1.5-1.5H3.75A1.5 1.5 0 0 0 2.25 6v12a1.5 1.5 0 0 0 1.5 1.5Zm10.5-11.25h.008v.008h-.008V8.25Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
  <div
    class="p-6 pb-4 pt-0"
  >
    <div
      class="block antialiased tracking-normal font-sans text-5xl font-semibold leading-tight text-inherit h-3 rounded-full bg-gray-300"
    >
       
    </div>
  </div>
</div>
`;

exports[`ContentCard skeleton > should render correctly with custom className 1`] = `
<div
  class="relative bg-clip-border text-gray-700 flex select-none flex-col overflow-hidden rounded-xl bg-white shadow-lg animate-pulse mt-4"
>
  <div
    class="bg-clip-border rounded-xl overflow-hidden text-gray-700 relative m-0 mb-5 grid h-64 place-items-center rounded-b-none bg-gray-300"
  >
    <svg
      aria-hidden="true"
      class="h-12 w-12 text-gray-500"
      data-slot="icon"
      fill="none"
      stroke="currentColor"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="m2.25 15.75 5.159-5.159a2.25 2.25 0 0 1 3.182 0l5.159 5.159m-1.5-1.5 1.409-1.409a2.25 2.25 0 0 1 3.182 0l2.909 2.909m-18 3.75h16.5a1.5 1.5 0 0 0 1.5-1.5V6a1.5 1.5 0 0 0-1.5-1.5H3.75A1.5 1.5 0 0 0 2.25 6v12a1.5 1.5 0 0 0 1.5 1.5Zm10.5-11.25h.008v.008h-.008V8.25Zm.375 0a.375.375 0 1 1-.75 0 .375.375 0 0 1 .75 0Z"
        stroke-linecap="round"
        stroke-linejoin="round"
      />
    </svg>
  </div>
  <div
    class="p-6 pb-4 pt-0"
  >
    <div
      class="block antialiased tracking-normal font-sans text-5xl font-semibold leading-tight text-inherit h-3 rounded-full bg-gray-300"
    >
       
    </div>
  </div>
</div>
`;
